akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -41,16 +42,38 @@ trait Projection[Envelope] {
   /**
    * Run the Projection.
    */
-  def run()(implicit systemProvider: ClassicActorSystemProvider): Unit
+  def run()(implicit systemProvider: ClassicActorSystemProvider): RunningProjection
 
-  def runWithBackoff()(implicit systemProvider: ClassicActorSystemProvider): Unit
+  def runWithBackoff()(implicit systemProvider: ClassicActorSystemProvider): RunningProjection
```

Comment:
Given what we decided for the RestartSource and KillSwitch, do we still need both `run` and `runWithBackoff`? If restarts should never be used from testkit we could make it possible to disable restarts via ProjectionSettings and the testkit would use that before `run`.

Reply:
I don't see how we could disable restarts via ProjectionSettings, unless we add more settings.

Reply:
do mean by changing it by defining a `maxRestarts = 0`?

The settings doesn't have restarts, we will need to add for the tests. I don't think that making that configurable makes sense for projections.

Reply:
yes, or a separate property that is taken into account when adding the RestartSource or not adding it.

Reply:
I'm adding `maxRestarts`. I prefer to avoid test flags. Ultimately, a user can disable the backoff by setting maxRestart to 0. 

I don't think it makes sense in practice, but between this and a flag for disabling the backoff, then better to support the full `RestartSource` API.